examples/nlp/machine_translation/nmt_server/server.py

```diff
@@ -0,0 +1,102 @@
+    parser.add_argument("--model", required=True, action='append', help="")
+    parser.add_argument("--port", default=50052, type=int, required=False)
+    parser.add_argument("--batch_size", type=int, default=256, help="")
+    parser.add_argument("--beam_size", type=int, default=1, help="")
+    parser.add_argument("--len_pen", type=float, default=0.6, help="")
+    parser.add_argument("--max_delta_length", type=int, default=5, help="")
+
+    args = parser.parse_args()
+    return args
+
+def batches(lst, n):
+    """Yield successive n-sized chunks from lst."""
+    for i in range(0, len(lst), n):
+        yield lst[i:i + n]
+
+class JarvisTranslateServicer(nmtsrv.JarvisTranslateServicer):
+    """Provides methods that implement functionality of route guide server."""
+
+    def __init__(self, model_paths, beam_size=1, len_pen=0.6, max_delta_length=5, batch_size=256):
+        self._models = {}
+        self._beam_size = beam_size
+        self._len_pen = len_pen
+        self._max_delta_length = max_delta_length
+        self._batch_size = batch_size
+
+        for model_path in model_paths:
+            logging.info(f"Loading model {model_path}")
+            self._load_model(model_path)
+        logging.info("Models loaded. Ready for inference requests.")
+    
+    def _load_model(self, model_path):
+        model_name, _ = os.path.splitext(os.path.basename(model_path))
+        model_name = model_name.lower()
+        if not re.match("^[\w]{2}-[\w]{2}$", model_name):
+            logging.error("Model not named in language pair format src-target")
+        if model_path.endswith(".nemo"):
+            logging.info("Attempting to initialize from .nemo file")
+            self._models[model_name] = nemo_nlp.models.machine_translation.MTEncDecModel.restore_from(restore_path=model_path)
+        else:
+            raise NotImplemented(f"Only support .nemo files, but got: {model_path}")
+
+        self._models[model_name].beam_search.beam_size = self._beam_size
+        self._models[model_name].beam_search.len_pen = self._len_pen
+        self._models[model_name].beam_search.max_delta_length = self._max_delta_length
+
+        if torch.cuda.is_available():
+            self._models[model_name] = self._models[model_name].cuda()
+
+    def TranslateText(self, request, context):
+        logging.info(f"Request received w/ {len(request.texts)} utterances")
+        results = []
+
+        lang_pair = f"{request.source_language}-{request.target_language}".lower()
+        if lang_pair not in self._models:
+            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+            context.set_details(f"Invalid lang pair {lang_pair} requeste")
+            return nmt.TranslateTextResponse()
+
+        request_strings = [x for x in request.texts]
+
+        for batch in batches(request_strings, self._batch_size):
+            batch_results = self._models[lang_pair].translate(text=batch, source_lang=None, target_lang=None)
```

Comment:
Is there a reason why source and target lang is None here? This won't use the correct input tokenization and target detokenization.

Reply:
Nope, no good reason. Better question is why doesn't the model already know the `source_lang` and `target_lang` if it matters? It's being restored from a nemo model.. that data should already be there.

Reply:
We've added that now and new models won't require this (#1805). But for existing models, and for temporary backward compatibility, the translate method has a source_lang and target_lang that need to be set.